api/dps/server.go

```diff
@@ -31,28 +32,29 @@ import (
 // This is generally an on-disk interface, but could be a GRPC-based index as
 // well, in which case there is a double redirection.
 type Server struct {
-	index dps.Reader
-	codec dps.Codec
-
+	index    dps.Reader
+	codec    dps.Codec
+	tracer   trace.Tracer
 	validate *validator.Validate
 }
 
 // NewServer creates a new server, using the provided index reader as a backend
 // for data retrieval.
-func NewServer(index dps.Reader, codec dps.Codec) *Server {
+func NewServer(index dps.Reader, codec dps.Codec, tracer trace.Tracer) *Server {
 
 	s := Server{
 		index:    index,
 		codec:    codec,
 		validate: validator.New(),
+		tracer:   tracer,
 	}
 
 	return &s
 }
 
 // GetFirst implements the `GetFirst` method of the generated GRPC server.
-func (s *Server) GetFirst(_ context.Context, _ *GetFirstRequest) (*GetFirstResponse, error) {
-
+func (s *Server) GetFirst(ctx context.Context, _ *GetFirstRequest) (*GetFirstResponse, error) {
+	s.tracer.StartSpanFromContext(ctx, trace.GetFirst)
```

Comment:
you would need to call `defer span.End()` to properly mark span as finished.

Reply:
And likely pass the `ctx` to the underlying function calls (at least for the more complex APIs with internal logic). Otherwise traces would be quite shallow with a single span whose duration is equal to the duration of the api call.

Reply:
I considered this,I'm only doing it for `GetRegisterValues` since that's the only API with a more complex underlying function. The rest are simple badger read/returns

Reply:
okie-dokie!  `GetRegisterValues` would likely need a change since it is quite complex inside and something tells me that some (all?) of the iterations can be quite expensive